src/cryptography/x509/ocsp.py

```diff
@@ -184,279 +137,339 @@ def __init__(
         """
         The responder's key hash or None
         """
 
     @abc.abstractproperty
-    def responder_name(self):
+    def responder_name(self) -> typing.Optional[x509.Name]:
         """
         The responder's Name or None
         """
 
     @abc.abstractproperty
-    def produced_at(self):
+    def produced_at(self) -> datetime.datetime:
         """
         The time the response was produced
         """
 
     @abc.abstractproperty
-    def certificate_status(self):
+    def certificate_status(self) -> OCSPCertStatus:
         """
         The status of the certificate (an element from the OCSPCertStatus enum)
         """
 
     @abc.abstractproperty
-    def revocation_time(self):
+    def revocation_time(self) -> typing.Optional[datetime.datetime]:
         """
         The date of when the certificate was revoked or None if not
         revoked.
         """
 
     @abc.abstractproperty
-    def revocation_reason(self):
+    def revocation_reason(self) -> typing.Optional[x509.ReasonFlags]:
         """
         The reason the certificate was revoked or None if not specified or
         not revoked.
         """
 
     @abc.abstractproperty
-    def this_update(self):
+    def this_update(self) -> datetime.datetime:
         """
         The most recent time at which the status being indicated is known by
         the responder to have been correct
         """
 
     @abc.abstractproperty
-    def next_update(self):
+    def next_update(self) -> typing.Optional[datetime.datetime]:
         """
         The time when newer information will be available
         """
 
     @abc.abstractproperty
-    def issuer_key_hash(self):
+    def issuer_key_hash(self) -> bytes:
         """
         The hash of the issuer public key
         """
 
     @abc.abstractproperty
-    def issuer_name_hash(self):
+    def issuer_name_hash(self) -> bytes:
         """
         The hash of the issuer name
         """
 
     @abc.abstractproperty
-    def hash_algorithm(self):
+    def hash_algorithm(self) -> hashes.HashAlgorithm:
         """
         The hash algorithm used in the issuer name and key hashes
         """
 
     @abc.abstractproperty
-    def serial_number(self):
+    def serial_number(self) -> int:
         """
         The serial number of the cert whose status is being checked
         """
 
     @abc.abstractproperty
-    def extensions(self):
+    def extensions(self) -> x509.Extensions:
         """
         The list of response extensions. Not single response extensions.
         """
 
     @abc.abstractproperty
-    def single_extensions(self):
+    def single_extensions(self) -> x509.Extensions:
         """
         The list of single response extensions. Not response extensions.
         """
+
+
+class OCSPRequestBuilder(object):
+    def __init__(self, request=None, extensions=[]):
+        self._request = request
+        self._extensions = extensions
+
+    def add_certificate(
```

Comment:
All these functions should have `-> "OCSPRequestBuilder"` for return type so mypy can do chaining.